### PR TITLE
Replaces selectSource with selectLocation

### PR DESCRIPTION
**Pull request by lukaszsobek**
Associated Issue: #4608

### Summary of Changes

* Replaces occurences of selectSource with selectLocation except in mochitests
* Reformats input to selectLocation to conform with the new requirements

### Test Plan
- yarn test
- yarn flow check

Still needs mochitest

### CODE REVIEW

**Comment by bomsy** [comment]
@lukaszsobek there is a `selectSource` in this file `devtools/client/debugger/new/test/mochitest/browser_dbg-console-link.js` causing the travis error. You might need to switch that to `selectLocation` as well.

**Comment by lukaszsobek** [comment]
@bomsy - some of these seem to call the mochitest version of selectSource (which is intentionally left as selectSource to distinguish it from selectLocation). Will get to the mochitest as soon as I get it installed.

File: src/actions/sources.js [hunk 1]
```diff
@@ -174,19 +174,28 @@ export function selectSourceURL(
...
+  column?: ?number,
+  sourceUrl?: ?string
+};
+
 /**
  * @memberof actions/sources
  * @static
  */
-export function selectSource(id: string, options: SelectSourceOptions = {}) {
+export function selectLocation(location: LocationObject) {
```
Comment (bomsy) [inline src/actions/sources.js]: @lukaszsobek for now don't replace this function as it is used in MC here 
https://dxr.mozilla.org/mozilla-central/source/devtools/client/shared/view-source.js#65
maybe just create a new `selectLocation` function and call `selectSource` from there

Comment (bomsy) [reply src/actions/sources.js]: Might have to change that eventually

File: assets/panel/panel.js [hunk 2]
```diff
@@ -98,8 +98,9 @@ DebuggerPanel.prototype = {
     return { frames, selected };
   },
 
-  selectSource(sourceURL, sourceLine) {
-    this._actions.selectSourceURL(sourceURL, { location: { line: sourceLine }});
+  selectLocation(sourceOptions) {
```
Comment (bomsy) [inline assets/panel/panel.js]: @lukaszsobek for now don't replace this function, as it is used in MC as well here 
https://dxr.mozilla.org/mozilla-central/source/devtools/client/shared/view-source.js#65.
Maybe just create a new `selectLocation` function and call `selectSource` from there.
That's probably the issue with the tests as well

File: src/actions/sources.js (hunk 1 above)
Comment (bomsy) [reply src/actions/sources.js]: @lukaszsobek sorry! ignore this comment here... i've move the comment up to the right location [here](https://github.com/devtools-html/debugger.html/pull/4789/files#r153234382)

**Comment by jasonLaster** [comment]
just pushed a rebase

**Comment by jasonLaster** [comment]
@lukaszsobek good news, i think everything is working. Lets see if we go green :)

**Comment by lukaszsobek** [comment]
All green :)

**Comment by jasonLaster** [comment]
This is amazing work. Thank you so much for helping us upgrade this API.

Changes like this are how we keep the code base a place we are happy to work in as opposed to letting it slip into something that we just shrug and accept. 

There is no way, that I could have found the time or energy to complete a migration of this size. So, I'm really grateful that you were able to do it. 

I hope you learned a bit along the way about how some of these components work.

**Comment by lukaszsobek** [comment]
It was so good I even wrote a blog post about it. An incredibly rewarding experience! Thank you.

**Comment by bomsy** [comment]
Nice one! Share your blogpost with us

**Pull request merged**